### PR TITLE
Add a test for a compiler crasher with -O -whole-module-optimization -sanitize=address

### PR DESCRIPTION
**Pull request by kubamracek**
Add a test for <https://github.com/apple/swift-llvm/pull/133>.

### CODE REVIEW

File: test/Interpreter/asan-globals-overwrite.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// RUN: %target-build-swift -O -whole-module-optimization -sanitize=address %s -o %t
+// RUN: %target-codesign %t
+// RUN: %target-run %t
```
Comment (rjmccall) [inline test/Interpreter/asan-globals-overwrite.swift]: I think this is `%target-run-simple-swift(-O -whole-module-optimization -sanitize=address)`.

**Comment by CodaFi** [comment]
Still good to have this test

@swift-ci please smoke test

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

File: test/Interpreter/asan-globals-overwrite.swift [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+// REQUIRES: executable_test
+// REQUIRES: objc_interop
```
Comment (yln) [inline test/Interpreter/asan-globals-overwrite.swift]: `REQUIRES: foundation`

**Comment by danliew-apple** [comment]
Interesting this looks like an old PR that didn't get merged.

File: test/Interpreter/asan-globals-overwrite.swift (hunk 2 above)
Comment (danliew-apple) [inline test/Interpreter/asan-globals-overwrite.swift]: Probably needs `REQUIRES: asan_runtime` if we are actually going to run.

**Review by ghost — CHANGES_REQUESTED** [review]
I don't know why this test is in `test/Interpreter/`, we aren't using the interpreter in this test are we?
It sound probably be in `test/Sanitizers` with the other sanitizer test cases.

**Comment by CodaFi** [comment]
Merged in #31008.

**Pull request closed without merging**